### PR TITLE
Add global error handler

### PR DESCRIPTION
**Pull request by JoaoVictor6**
I create a simple error boundary for capturing exceptions and display a generic toast as mentioned in issue #26

Above a demo video:

https://github.com/ocodista/trampar-de-casa/assets/68869379/e9417dd2-f0ba-41a7-aa8d-ab5b5b628974

### Some questions
- The message is good?
- Make sense to create a generic fallback UI?

### CODE REVIEW

File: apps/web/prisma/client/index.js [hunk 1]
```diff
@@ -119,7 +119,7 @@ const config = {
       "value": "prisma-client-js"
     },
     "output": {
-      "value": "C:\\Users\\mateus.lima\\Documents\\personal\\dev\\JavaScript\\trampar-de-casa\\apps\\web\\prisma\\client",
+      "value": "/home/joao/FOSS/trampar-de-casa/apps/web/prisma/client",
```
Comment (JoaoVictor6) [inline apps/web/prisma/client/index.js:122]: This file is necessary?

File: apps/web/global/GlobalErrorBoundary.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      title: "Oops",
+      variant: "destructive",
+      description: 'Algo deu errado 🥶',
+    });
+  }
+
+  render() {
+    return this.props.children;
+  }
+}
```
Comment (ocodista) [inline apps/web/global/GlobalErrorBoundary.tsx:28]: Is there a way to do this with react hooks?

Also, as we're using next, maybe we could use the suggested error handler https://nextjs.org/docs/app/building-your-application/routing/error-handling

You could try this by throwing an error on the click of the button of the Hero section

**Review by ocodista — CHANGES_REQUESTED** [review]

**Comment by JoaoVictor6** [comment]
# Some chances

https://github.com/ocodista/trampar-de-casa/assets/68869379/66e86038-ce4a-44b4-bb00-b20aa7f201c9

- Using nextjs error handling approach
- Add a shadcn button for fallback UI

Valid @ocodista ? maybe fallback UI is not necessary

**Comment by ocodista** [comment]
João, peço desculpas mas acabei implementando o component global-error.tsx na pasta root que já faz esse papel.

Se quiser, você pode tentar pegar essa estilização de botão (ou página) e colocar nele, aí eu aprovo o PR, o que vc acha?

**Comment by JoaoVictor6** [comment]
> João, peço desculpas mas acabei implementando o component global-error.tsx na pasta root que já faz esse papel.
> 
> Se quiser, você pode tentar pegar essa estilização de botão (ou página) e colocar nele, aí eu aprovo o PR, o que vc acha?

Opa, tudo bem :)

**Comment by ocodista** [comment]
Closing for now.

If you're able to enhance the global-error page, please open a new PR.

Thank you!

**Pull request closed without merging**